### ISSUE TITLE
Unexpected error while processing request: undefined method `extract_multipart' for Rack::Multipart:Module

### ISSUE THREAD

**Issue by bcoles** [beefproject/beef#1628]
```
I, [2018-10-01T11:42:35.547236 #8899]  INFO -- : Thin web server (v1.7.2 codename Bachmanity)
D, [2018-10-01T11:42:35.547318 #8899] DEBUG -- : Debugging ON
I, [2018-10-01T11:42:35.547380 #8899]  INFO -- : Maximum connections set to 1024
I, [2018-10-01T11:42:35.547436 #8899]  INFO -- : Listening on 0.0.0.0:3000, CTRL+C to stop
172.16.191.187 - - [01/Oct/2018:11:42:35 -0400] "GET /ui/authentication HTTP/1.1" 200 1028 0.0014
172.16.191.187 - - [01/Oct/2018:11:42:35 -0400] "GET /ui/authentication HTTP/1.1" 200 1028 0.0013
172.16.191.187 - - [01/Oct/2018:11:42:35 -0400] "GET /ui/authentication HTTP/1.1" 200 1028 0.0012
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/javascript/ext-base.js HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/javascript/ext-all.js HTTP/1.1" 304 - 0.0001
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/javascript-min/web_ui_auth.js HTTP/1.1" 200 1182 0.0003
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/css/ext-all.css HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/beef.png HTTP/1.1" 304 - 0.0005
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/default/panel/corners-sprite.gif HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/default/panel/top-bottom.gif HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/default/panel/left-right.gif HTTP/1.1" 304 - 0.0001
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/default/button/btn.gif HTTP/1.1" 304 - 0.0018
172.16.191.187 - - [01/Oct/2018:11:42:36 -0400] "GET /ui/media/images/default/form/text-bg.gif HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:38 -0400] "GET /ui/media/images/default/box/tb-blue.gif HTTP/1.1" 304 - 0.0002
172.16.191.187 - - [01/Oct/2018:11:42:38 -0400] "GET /ui/media/images/default/grid/loading.gif HTTP/1.1" 304 - 0.0003
E, [2018-10-01T11:42:38.684765 #8899] ERROR -- : Unexpected error while processing request: undefined method `extract_multipart' for Rack::Multipart:Module
	/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/request.rb:472:in `parse_multipart'
	/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/request.rb:335:in `POST'
	/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/request.rb:358:in `params'
	/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/request.rb:20:in `params'
	/home/user/Desktop/beef/extensions/admin_ui/classes/httpcontroller.rb:42:in `run'
	/home/user/Desktop/beef/extensions/admin_ui/handlers/ui.rb:31:in `call'
	/usr/lib/ruby/vendor_ruby/rack/urlmap.rb:66:in `block in call'
	/usr/lib/ruby/vendor_ruby/rack/urlmap.rb:50:in `each'
	/usr/lib/ruby/vendor_ruby/rack/urlmap.rb:50:in `call'
	/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/common_logger.rb:33:in `call'
	/var/lib/gems/2.5.0/gems/sinatra-2.0.4/lib/sinatra/base.rb:231:in `call'
	/usr/lib/ruby/vendor_ruby/thin/connection.rb:86:in `block in pre_process'
	/usr/lib/ruby/vendor_ruby/thin/connection.rb:84:in `catch'
	/usr/lib/ruby/vendor_ruby/thin/connection.rb:84:in `pre_process'
	/usr/lib/ruby/vendor_ruby/thin/connection.rb:53:in `process'
	/usr/lib/ruby/vendor_ruby/thin/connection.rb:39:in `receive_data'
	/var/lib/gems/2.5.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run_machine'
	/var/lib/gems/2.5.0/gems/eventmachine-1.0.9.1/lib/eventmachine.rb:193:in `run'
	/usr/lib/ruby/vendor_ruby/thin/backends/base.rb:73:in `start'
	/usr/lib/ruby/vendor_ruby/thin/server.rb:162:in `start'
	/home/user/Desktop/beef/core/main/server.rb:161:in `start'
	./beef:231:in `<main>'

```

**Comment by xfox64x** [comment]
I don't know much about Ruby and I don't think the solution I have is a good one, but (if one is using Ruby 2.5 on Kali) one could go back into Gemfile, revert the versions of rack and sinatra back to 2.0, delete the Gemfile.lock, rerun ```bundle install```, and then open up the offending ```/var/lib/gems/2.5.0/gems/rack-2.0.5/lib/rack/request.rb``` and add a nice little ```require 'rack/multipart'``` after the first two ```require``` lines. 

Be aware that, if you've attempted this recently, your rack version is probably 2.0.6, so your path would be ```/var/lib/gems/2.5.0/gems/rack-2.0.6/lib/rack/request.rb````. Also, I have like 10 minutes of experience in Ruby, so I don't know if this is how you fix things like this... but it works. I'll open an issue with rack.

**Comment by TheFrenchGuy** [comment]
Thank you very much this did fix indeed the problem on parrot os that I wasn't able to login into beef-xss

**Comment by MaatRa** [comment]
@xfox64x thank you for the info, it worked perfectly with rack-2.0.6.

**Comment by bcoles** [comment]
@xfox64x  Thanks for the patch and issue: https://github.com/rack/rack/issues/1315

**Comment by bcoles** [comment]
**TL/DR: At this stage, the best advice I can offer is to run:**

```
sudo apt-get update
sudo apt-get dist-upgrade
sudo apt-get install ruby-rack ruby-rack-protection ruby-sinatra
```

---

Issues relating to Rack have been reported frequently over the past year or so. This particular issue, and others which I suspect are related, are difficult to reproduce.

Compounding the issues was the HTTP handler, which was silently swallowing stack traces. I've fixed that in 214e5b977f49f9775db34c8733e99350421571c0, so at least there's some visibility into the issues errors, without having to enable `beef.http.debug` mode.

In the past, I have been able to reproduce these issues, but not on a fully updated system built from latest ISOs.

I cannot reproduce these issues with BeEF from `master` on latest and fully updated Kali and Parrot.

Today, I started with an old Parrot 3.1.1 system, which was having this issue (`undefined method `extract_multipart' for Rack::Multipart:Module`). The system had been updated since install, so it wasn't a fresh 3.1.1 system, but the packages also hadn't been updated in 6 months or so, so it seemed like a feasible test system.

I updated BeEF to the latest stable version from `master` and confirmed the issue was still present.

**I resolved the issue by running the following:**

```sh
apt-get install ruby2.5 ruby2.5-dev ruby2.5-doc
apt-get install ruby-rack
```

The output is shown below. Note that several other required dependencies were also installed/updated as part of this process:

```
The following NEW packages will be installed:
  libgdbm6
The following packages will be upgraded:
  libgdbm-compat4 libruby2.5 libssl-dev libssl1.1 ruby2.5 ruby2.5-dev ruby2.5-doc
```

```
The following NEW packages will be installed:
  ruby-mustermann
The following packages will be upgraded:
  ruby-rack ruby-rack-protection ruby-sinatra
```

```
─╼ #apt-get install ruby2.5 ruby2.5-dev ruby2.5-doc
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  abiword abiword-common abiword-plugin-grammar couchdb cpufrequtils crafty
  crafty-books-medtosmall dh-python dmz-cursor-theme emacs25-nox erlang17-asn1 erlang17-base
  erlang17-crypto erlang17-eunit erlang17-inets erlang17-mnesia erlang17-os-mon
  erlang17-public-key erlang17-runtime-tools erlang17-snmp erlang17-ssl erlang17-syntax-tools
  erlang17-tools erlang17-webtool erlang17-xmerl evolution-data-server-common ffmpegthumbnailer
  fonts-open-sans fruit geoip-database-extra gir1.2-mate-desktop glassfish-javaee
  gnome-themes-standard gnumeric gnumeric-common gnumeric-doc irssi libabiword-3.0 libairspyhf0
  libann0 libapache2-mod-dnssd libargon2-0 libarmadillo8 libavdevice57 libavfilter6 libavformat57
  libavresample3 libbabeltrace-ctf1 libbotan-1.10-1 libcamel-1.2-60 libcamel-1.2-62 libcdio17
  libchamplain-0.12-0 libchamplain-gtk-0.12-0 libclutter-1.0-0 libclutter-1.0-common
  libclutter-gtk-1.0-0 libcodec2-0.7 libcogl-common libcogl-pango20 libcogl-path20 libcogl20
  libdns-export1100 libdns169 libebackend-1.2-10 libebook-1.2-19 libebook-contacts-1.2-2
  libecal-1.2-19 libedata-book-1.2-25 libedataserver-1.2-22 libedataserver-1.2-23 libfabric1
  libffmpegthumbnailer4v5 libfile-copy-recursive-perl libfreerdp-cache1.1 libfreerdp-client1.1
  libfreerdp-codec1.1 libfreerdp-common1.1.0 libfreerdp-core1.1 libfreerdp-crypto1.1
  libfreerdp-gdi1.1 libfreerdp-locale1.1 libfreerdp-primitives1.1 libfreerdp-utils1.1
  libgcr-3-common libgdcm2.8 libgeoclue-2-0 libgeos-3.6.2 libgl2ps1.4 libgles2-mesa libgltf-0.1-1
  libgnome-desktop-3-12 libgnuradio-analog3.7.11 libgnuradio-atsc3.7.11 libgnuradio-audio3.7.11
  libgnuradio-blocks3.7.11 libgnuradio-channels3.7.11 libgnuradio-comedi3.7.11
  libgnuradio-digital3.7.11 libgnuradio-dtv3.7.11 libgnuradio-fcd3.7.11 libgnuradio-fec3.7.11
  libgnuradio-fft3.7.11 libgnuradio-filter3.7.11 libgnuradio-fosphor3.7.11 libgnuradio-noaa3.7.11
  libgnuradio-pager3.7.11 libgnuradio-pmt3.7.11 libgnuradio-qtgui3.7.11 libgnuradio-runtime3.7.11
  libgnuradio-trellis3.7.11 libgnuradio-uhd3.7.11 libgnuradio-video-sdl3.7.11
  libgnuradio-vocoder3.7.11 libgnuradio-wavelet3.7.11 libgnuradio-wxgui3.7.11
  libgnuradio-zeromq3.7.11 libgoffice-0.10-10 libgoffice-0.10-10-common libhdf5-openmpi-100
  libhttp-parser2.7.1 libhttp-server-simple-perl libisc166 libisl15 libjs-excanvas
  libjs-jquery-easing libjs-jquery-fancybox libjs-jquery-form libjs-jquery-mousewheel
  libjs-openlayers libjs-source-map libjs-uglify liblept5 liblimesuite17.12-1 liblink-grammar5
  liblivemedia61 liblivemedia62 libllvm5.0 libloudmouth1-0 libmagickcore-6.q16-3
  libmagickcore-6.q16-3-extra libmagickcore-6.q16-5 libmagickcore-6.q16-5-extra
  libmagickwand-6.q16-3 libmagickwand-6.q16-5 libmozjs185-1.0 libmpfr4 libnetcdf-c++4 libnfs8
  libopencv-calib3d3.2 libopencv-contrib3.2 libopencv-features2d3.2 libopencv-flann3.2
  libopencv-highgui3.2 libopencv-imgcodecs3.2 libopencv-ml3.2 libopencv-objdetect3.2
  libopencv-photo3.2 libopencv-shape3.2 libopencv-stitching3.2 libopencv-superres3.2
  libopencv-video3.2 libopencv-videoio3.2 libopencv-videostab3.2 libopencv-viz3.2 libopenexr22
  libopenmpi2 libopenmpi3 libopenobex2 liborcus-0.12-0 libots0 libphonenumber7 libplacebo2
  libplacebo4 libpmix2 libpoppler72 libpoppler73 libpostproc54 libproj12 libpsm-infinipath1
  libpsm2-2 libqt5designercomponents5 libqt5multimediaquick-p5 libqt5quicktest5 libqwt5-qt4
  libradare2-2.1 libradare2-2.4 librte-acl2 librte-cfgfile2 librte-cmdline2 librte-cryptodev2
  librte-distributor1 librte-eal3 librte-ethdev5 librte-hash2 librte-ip-frag1 librte-jobstats1
  librte-kni2 librte-kvargs1 librte-lpm2 librte-mbuf2 librte-mempool2 librte-meter1 librte-net1
  librte-pdump1 librte-pipeline3 librte-pmd-af-packet1 librte-pmd-bnxt1 librte-pmd-bond1
  librte-pmd-cxgbe1 librte-pmd-e1000-1 librte-pmd-ena1 librte-pmd-enic1 librte-pmd-fm10k1
  librte-pmd-i40e1 librte-pmd-ixgbe1 librte-pmd-null-crypto1 librte-pmd-null1 librte-pmd-pcap1
  librte-pmd-qede1 librte-pmd-ring2 librte-pmd-thunderx-nicvf1 librte-pmd-vhost1
  librte-pmd-virtio1 librte-pmd-vmxnet3-uio1 librte-pmd-xenvirt1 librte-port3 librte-power1
  librte-reorder1 librte-ring1 librte-sched1 librte-table2 librte-timer1 librte-vhost3 libruby2.3
  libsctp1 libsndio6.1 libsocket++1 libsrtp0 libswscale4 libsynctex1 libtcl8.5 libtelepathy-glib0
  libtesseract4 libtk8.5 libuhd003.010.002 libuhd003.010.003 libvolk1.3 libvpx4 libvtk6.3
  libwayland-egl1-mesa libwinpr-crt0.1 libwinpr-crypto0.1 libwinpr-dsparse0.1
  libwinpr-environment0.1 libwinpr-file0.1 libwinpr-handle0.1 libwinpr-heap0.1 libwinpr-input0.1
  libwinpr-interlocked0.1 libwinpr-library0.1 libwinpr-path0.1 libwinpr-pool0.1
  libwinpr-registry0.1 libwinpr-rpc0.1 libwinpr-sspi0.1 libwinpr-synch0.1 libwinpr-sysinfo0.1
  libwinpr-thread0.1 libwinpr-utils0.1 libwireshark10 libwiretap7 libwscodecs1 libwsutil8
  libwv-1.2-4 libwww-mechanize-perl libx265-146 libxfont1 link-grammar-dictionaries-en
  linux-headers-4.14.0-parrot13-amd64 linux-headers-4.14.0-parrot13-common
  linux-image-4.14.0-parrot13-amd64 linux-kbuild-4.14 mate-backgrounds mate-user-share
  mate-user-share-common minisat obex-data-server openmpi-bin openmpi-common php7.0-mysql polyglot
  pxlib1 python-functools32 python-ply python-pycryptodome python-pysmi python-pysnmp4
  python-pysnmp4-apps python-pysnmp4-mibs python-unicodecsv qbs-common qml-module-qtqml-models2
  qmlscene qt3d5-doc qt5-assistant qt5-doc qtbase5-dev-tools qtbase5-doc qtcharts5-doc
  qtconnectivity5-doc qtcreator-data qtcreator-doc qtdeclarative5-dev-tools qtdeclarative5-doc
  qtgraphicaleffects5-doc qtlocation5-doc qtmultimedia5-doc qtquickcontrols2-5-doc
  qtquickcontrols5-doc qtscript5-doc qtsensors5-doc qtserialport5-doc qtsvg5-doc qttools5-doc
  qtvirtualkeyboard5-doc qtwayland5-doc qtwebchannel5-doc qtwebengine5-doc qtwebkit5-doc
  qtwebkit5-examples-doc qtwebsockets5-doc qtwebview5-doc qtx11extras5-doc
  qtxmlpatterns5-dev-tools qtxmlpatterns5-doc ruby-ansi ruby-celluloid ruby-celluloid-io
  ruby-colorize ruby-daemons ruby-dataobjects ruby-dataobjects-mysql ruby-dataobjects-postgres
  ruby-dataobjects-sqlite3 ruby-diff-lcs ruby-dm-core ruby-dm-do-adapter ruby-dm-migrations
  ruby-dm-serializer ruby-dm-sqlite-adapter ruby-docile ruby-em-proxy ruby-em-websocket
  ruby-erubis ruby-execjs ruby-geoip ruby-hitimes ruby-libv8 ruby-mojo-magick ruby-msgpack
  ruby-net-dns ruby-network-interface ruby-nio4r ruby-packetfu ruby-parseconfig ruby-pcaprub
  ruby-qr4r ruby-rack ruby-rack-protection ruby-ref ruby-rqrcode ruby-rspec-expectations
  ruby-rspec-support ruby-rubydns ruby-simplecov ruby-simplecov-html ruby-sinatra
  ruby-term-ansicolor ruby-therubyracer ruby-tilt ruby-timers ruby-tins ruby-uglifier ruby2.3
  ruby2.3-dev smitools thin tk8.5 xul-ext-noscript xul-ext-useragentswitcher
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  libgdbm-compat4 libgdbm6 libruby2.5 libssl-dev libssl1.1
Suggested packages:
  gdbm-l10n
The following NEW packages will be installed:
  libgdbm6
The following packages will be upgraded:
  libgdbm-compat4 libruby2.5 libssl-dev libssl1.1 ruby2.5 ruby2.5-dev ruby2.5-doc
7 upgraded, 1 newly installed, 0 to remove and 2126 not upgraded.
Need to get 9,794 kB of archives.
After this operation, 1,634 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 libgdbm6 amd64 1.18.1-2 [64.5 kB]
Get:3 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 ruby2.5-dev amd64 2.5.3-3 [407 kB]
Get:5 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 libssl1.1 amd64 1.1.1a-1 [1,527 kB]     
Get:2 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 libgdbm-compat4 amd64 1.18.1-2 [44.0 kB]
Get:6 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 libruby2.5 amd64 2.5.3-3 [3,429 kB]     
Get:4 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 libssl-dev amd64 1.1.1a-1 [1,788 kB]
Get:7 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 ruby2.5 amd64 2.5.3-3 [393 kB]          
Get:8 http://mirror.lagoon.nc/pub/parrot parrot/main amd64 ruby2.5-doc all 2.5.3-3 [2,142 kB]      
Fetched 9,794 kB in 18s (545 kB/s)                                                                 
Reading changelogs... Done
apt-listchanges: Mailing root: apt-listchanges: news for parrot
Preconfiguring packages ...
Selecting previously unselected package libgdbm6:amd64.
(Reading database ... 512949 files and directories currently installed.)
Preparing to unpack .../0-libgdbm6_1.18.1-2_amd64.deb ...
Unpacking libgdbm6:amd64 (1.18.1-2) ...
Preparing to unpack .../1-libgdbm-compat4_1.18.1-2_amd64.deb ...
Unpacking libgdbm-compat4:amd64 (1.18.1-2) over (1.14.1-6+b1) ...
Preparing to unpack .../2-ruby2.5-dev_2.5.3-3_amd64.deb ...
Unpacking ruby2.5-dev:amd64 (2.5.3-3) over (2.5.1-5) ...
Preparing to unpack .../3-libssl-dev_1.1.1a-1_amd64.deb ...
Unpacking libssl-dev:amd64 (1.1.1a-1) over (1.1.0h-4) ...
Preparing to unpack .../4-libssl1.1_1.1.1a-1_amd64.deb ...
Unpacking libssl1.1:amd64 (1.1.1a-1) over (1.1.0h-4) ...
Preparing to unpack .../5-libruby2.5_2.5.3-3_amd64.deb ...
Unpacking libruby2.5:amd64 (2.5.3-3) over (2.5.1-5) ...
Preparing to unpack .../6-ruby2.5_2.5.3-3_amd64.deb ...
Unpacking ruby2.5 (2.5.3-3) over (2.5.1-5) ...
Preparing to unpack .../7-ruby2.5-doc_2.5.3-3_all.deb ...
Unpacking ruby2.5-doc (2.5.3-3) over (2.5.1-5) ...
Setting up libgdbm6:amd64 (1.18.1-2) ...
Processing triggers for libc-bin (2.27-6) ...
Setting up libssl1.1:amd64 (1.1.1a-1) ...
Processing triggers for man-db (2.8.4-2) ...
Setting up ruby2.5-doc (2.5.3-3) ...
Setting up libgdbm-compat4:amd64 (1.18.1-2) ...
Setting up libruby2.5:amd64 (2.5.3-3) ...
Setting up libssl-dev:amd64 (1.1.1a-1) ...
Setting up ruby2.5-dev:amd64 (2.5.3-3) ...
Setting up ruby2.5 (2.5.3-3) ...
Processing triggers for libc-bin (2.27-6) ...
Configuring sandbox profiles....
Sandbox profiles updated!
┌─[root@parrot]─[/home/user/Desktop/beef]
└──╼ #apt-get install ruby-rack
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  abiword abiword-common abiword-plugin-grammar couchdb cpufrequtils crafty
  crafty-books-medtosmall dh-python dmz-cursor-theme emacs25-nox erlang17-asn1 erlang17-base
  erlang17-crypto erlang17-eunit erlang17-inets erlang17-mnesia erlang17-os-mon
  erlang17-public-key erlang17-runtime-tools erlang17-snmp erlang17-ssl erlang17-syntax-tools
  erlang17-tools erlang17-webtool erlang17-xmerl evolution-data-server-common ffmpegthumbnailer
  fonts-open-sans fruit geoip-database-extra gir1.2-mate-desktop glassfish-javaee
  gnome-themes-standard gnumeric gnumeric-common gnumeric-doc irssi libabiword-3.0 libairspyhf0
  libann0 libapache2-mod-dnssd libargon2-0 libarmadillo8 libavdevice57 libavfilter6 libavformat57
  libavresample3 libbabeltrace-ctf1 libbotan-1.10-1 libcamel-1.2-60 libcamel-1.2-62 libcdio17
  libchamplain-0.12-0 libchamplain-gtk-0.12-0 libclutter-1.0-0 libclutter-1.0-common
  libclutter-gtk-1.0-0 libcodec2-0.7 libcogl-common libcogl-pango20 libcogl-path20 libcogl20
  libdns-export1100 libdns169 libebackend-1.2-10 libebook-1.2-19 libebook-contacts-1.2-2
  libecal-1.2-19 libedata-book-1.2-25 libedataserver-1.2-22 libedataserver-1.2-23 libfabric1
  libffmpegthumbnailer4v5 libfile-copy-recursive-perl libfreerdp-cache1.1 libfreerdp-client1.1
  libfreerdp-codec1.1 libfreerdp-common1.1.0 libfreerdp-core1.1 libfreerdp-crypto1.1
  libfreerdp-gdi1.1 libfreerdp-locale1.1 libfreerdp-primitives1.1 libfreerdp-utils1.1
  libgcr-3-common libgdcm2.8 libgeoclue-2-0 libgeos-3.6.2 libgl2ps1.4 libgles2-mesa libgltf-0.1-1
  libgnome-desktop-3-12 libgnuradio-analog3.7.11 libgnuradio-atsc3.7.11 libgnuradio-audio3.7.11
  libgnuradio-blocks3.7.11 libgnuradio-channels3.7.11 libgnuradio-comedi3.7.11
  libgnuradio-digital3.7.11 libgnuradio-dtv3.7.11 libgnuradio-fcd3.7.11 libgnuradio-fec3.7.11
  libgnuradio-fft3.7.11 libgnuradio-filter3.7.11 libgnuradio-fosphor3.7.11 libgnuradio-noaa3.7.11
  libgnuradio-pager3.7.11 libgnuradio-pmt3.7.11 libgnuradio-qtgui3.7.11 libgnuradio-runtime3.7.11
  libgnuradio-trellis3.7.11 libgnuradio-uhd3.7.11 libgnuradio-video-sdl3.7.11
  libgnuradio-vocoder3.7.11 libgnuradio-wavelet3.7.11 libgnuradio-wxgui3.7.11
  libgnuradio-zeromq3.7.11 libgoffice-0.10-10 libgoffice-0.10-10-common libhdf5-openmpi-100
  libhttp-parser2.7.1 libhttp-server-simple-perl libisc166 libisl15 libjs-excanvas
  libjs-jquery-easing libjs-jquery-fancybox libjs-jquery-form libjs-jquery-mousewheel
  libjs-openlayers libjs-source-map libjs-uglify liblept5 liblimesuite17.12-1 liblink-grammar5
  liblivemedia61 liblivemedia62 libllvm5.0 libloudmouth1-0 libmagickcore-6.q16-3
  libmagickcore-6.q16-3-extra libmagickcore-6.q16-5 libmagickcore-6.q16-5-extra
  libmagickwand-6.q16-3 libmagickwand-6.q16-5 libmozjs185-1.0 libmpfr4 libnetcdf-c++4 libnfs8
  libopencv-calib3d3.2 libopencv-contrib3.2 libopencv-features2d3.2 libopencv-flann3.2
  libopencv-highgui3.2 libopencv-imgcodecs3.2 libopencv-ml3.2 libopencv-objdetect3.2
  libopencv-photo3.2 libopencv-shape3.2 libopencv-stitching3.2 libopencv-superres3.2
  libopencv-video3.2 libopencv-videoio3.2 libopencv-videostab3.2 libopencv-viz3.2 libopenexr22
  libopenmpi2 libopenmpi3 libopenobex2 liborcus-0.12-0 libots0 libphonenumber7 libplacebo2
  libplacebo4 libpmix2 libpoppler72 libpoppler73 libpostproc54 libproj12 libpsm-infinipath1
  libpsm2-2 libqt5designercomponents5 libqt5multimediaquick-p5 libqt5quicktest5 libqwt5-qt4
  libradare2-2.1 libradare2-2.4 librte-acl2 librte-cfgfile2 librte-cmdline2 librte-cryptodev2
  librte-distributor1 librte-eal3 librte-ethdev5 librte-hash2 librte-ip-frag1 librte-jobstats1
  librte-kni2 librte-kvargs1 librte-lpm2 librte-mbuf2 librte-mempool2 librte-meter1 librte-net1
  librte-pdump1 librte-pipeline3 librte-pmd-af-packet1 librte-pmd-bnxt1 librte-pmd-bond1
  librte-pmd-cxgbe1 librte-pmd-e1000-1 librte-pmd-ena1 librte-pmd-enic1 librte-pmd-fm10k1
  librte-pmd-i40e1 librte-pmd-ixgbe1 librte-pmd-null-crypto1 librte-pmd-null1 librte-pmd-pcap1
  librte-pmd-qede1 librte-pmd-ring2 librte-pmd-thunderx-nicvf1 librte-pmd-vhost1
  librte-pmd-virtio1 librte-pmd-vmxnet3-uio1 librte-pmd-xenvirt1 librte-port3 librte-power1
  librte-reorder1 librte-ring1 librte-sched1 librte-table2 librte-timer1 librte-vhost3 libruby2.3
  libsctp1 libsndio6.1 libsocket++1 libsrtp0 libswscale4 libsynctex1 libtcl8.5 libtelepathy-glib0
  libtesseract4 libtk8.5 libuhd003.010.002 libuhd003.010.003 libvolk1.3 libvpx4 libvtk6.3
  libwayland-egl1-mesa libwinpr-crt0.1 libwinpr-crypto0.1 libwinpr-dsparse0.1
  libwinpr-environment0.1 libwinpr-file0.1 libwinpr-handle0.1 libwinpr-heap0.1 libwinpr-input0.1
  libwinpr-interlocked0.1 libwinpr-library0.1 libwinpr-path0.1 libwinpr-pool0.1
  libwinpr-registry0.1 libwinpr-rpc0.1 libwinpr-sspi0.1 libwinpr-synch0.1 libwinpr-sysinfo0.1
  libwinpr-thread0.1 libwinpr-utils0.1 libwireshark10 libwiretap7 libwscodecs1 libwsutil8
  libwv-1.2-4 libwww-mechanize-perl libx265-146 libxfont1 link-grammar-dictionaries-en
  linux-headers-4.14.0-parrot13-amd64 linux-headers-4.14.0-parrot13-common
  linux-image-4.14.0-parrot13-amd64 linux-kbuild-4.14 mate-backgrounds mate-user-share
  mate-user-share-common minisat obex-data-server openmpi-bin openmpi-common php7.0-mysql polyglot
  pxlib1 python-functools32 python-ply python-pycryptodome python-pysmi python-pysnmp4
  python-pysnmp4-apps python-pysnmp4-mibs python-unicodecsv qbs-common qml-module-qtqml-models2
  qmlscene qt3d5-doc qt5-assistant qt5-doc qtbase5-dev-tools qtbase5-doc qtcharts5-doc
  qtconnectivity5-doc qtcreator-data qtcreator-doc qtdeclarative5-dev-tools qtdeclarative5-doc
  qtgraphicaleffects5-doc qtlocation5-doc qtmultimedia5-doc qtquickcontrols2-5-doc
  qtquickcontrols5-doc qtscript5-doc qtsensors5-doc qtserialport5-doc qtsvg5-doc qttools5-doc
  qtvirtualkeyboard5-doc qtwayland5-doc qtwebchannel5-doc qtwebengine5-doc qtwebkit5-doc
  qtwebkit5-examples-doc qtwebsockets5-doc qtwebview5-doc qtx11extras5-doc
  qtxmlpatterns5-dev-tools qtxmlpatterns5-doc ruby-ansi ruby-celluloid ruby-celluloid-io
  ruby-colorize ruby-daemons ruby-dataobjects ruby-dataobjects-mysql ruby-dataobjects-postgres
  ruby-dataobjects-sqlite3 ruby-diff-lcs ruby-dm-core ruby-dm-do-adapter ruby-dm-migrations
  ruby-dm-serializer ruby-dm-sqlite-adapter ruby-docile ruby-em-proxy ruby-em-websocket
  ruby-erubis ruby-execjs ruby-geoip ruby-hitimes ruby-libv8 ruby-mojo-magick ruby-msgpack
  ruby-mustermann ruby-net-dns ruby-network-interface ruby-nio4r ruby-packetfu ruby-parseconfig
  ruby-pcaprub ruby-qr4r ruby-rack-protection ruby-ref ruby-rqrcode ruby-rspec-expectations
  ruby-rspec-support ruby-rubydns ruby-simplecov ruby-simplecov-html ruby-sinatra
  ruby-term-ansicolor ruby-therubyracer ruby-tilt ruby-timers ruby-tins ruby-uglifier ruby2.3
  ruby2.3-dev smitools thin tk8.5 xul-ext-noscript xul-ext-useragentswitcher
Use 'apt autoremove' to remove them.
The following additional packages will be installed:
  ruby-mustermann ruby-rack-protection ruby-sinatra
The following NEW packages will be installed:
  ruby-mustermann
The following packages will be upgraded:
  ruby-rack ruby-rack-protection ruby-sinatra
3 upgraded, 1 newly installed, 0 to remove and 2123 not upgraded.
Need to get 500 kB of archives.
After this operation, 205 kB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 ruby-mustermann all 1.0.0-4 [36.3 kB]
Get:2 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 ruby-rack-protection all 2.0.5-4 [36.4 kB]
Get:3 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 ruby-sinatra all 2.0.5-4 [337 kB]
Get:4 http://kartolo.sby.datautama.net.id/parrot parrot/main amd64 ruby-rack all 2.0.6-3 [90.0 kB] 
Fetched 500 kB in 7s (72.8 kB/s)                                                                   
Reading changelogs... Done
Selecting previously unselected package ruby-mustermann.
(Reading database ... 512984 files and directories currently installed.)
Preparing to unpack .../ruby-mustermann_1.0.0-4_all.deb ...
Unpacking ruby-mustermann (1.0.0-4) ...
Preparing to unpack .../ruby-rack-protection_2.0.5-4_all.deb ...
Unpacking ruby-rack-protection (2.0.5-4) over (1.5.3-2.1) ...
Preparing to unpack .../ruby-sinatra_2.0.5-4_all.deb ...
Unpacking ruby-sinatra (2.0.5-4) over (1.4.8-1) ...
Preparing to unpack .../ruby-rack_2.0.6-3_all.deb ...
Unpacking ruby-rack (2.0.6-3) over (1.6.4-5) ...
Setting up ruby-rack (2.0.6-3) ...
Setting up ruby-rack-protection (2.0.5-4) ...
Processing triggers for man-db (2.8.4-2) ...
Setting up ruby-mustermann (1.0.0-4) ...
Setting up ruby-sinatra (2.0.5-4) ...
Configuring sandbox profiles....
Sandbox profiles updated!
┌─[root@parrot]─[/home/user/Desktop/beef]
└──╼ #
```

I'm not sure there's much I can do about this, beyond hoping that the OS maintainers update their system packages, or that a new version of Rack is released that somehow mitigates the issues introduced by having old versions of `ruby-rack` or `sinatra` packages installed. It's possible that this issue could be resolved by adding `ruby-rack` and `ruby-sinatra` to the `install` installer.

Unfortunately it's not something I can easily test either, as building and re-building old versions of Kali and Parrot from older ISOs, then testing which exact combination of dependencies is causing issues, is tedious.

**Comment by bcoles** [comment]
Try this:

```
sudo apt-get update
sudo apt-get dist-upgrade
sudo apt-get install ruby-rack ruby-rack-protection ruby-sinatra
```

As per: https://github.com/beefproject/beef/issues/1628#issuecomment-464289170

**Comment by cleartany** [comment]
> 
> 
> Try this:
> 
> ```
> sudo apt-get update
> sudo apt-get dist-upgrade
> sudo apt-get install ruby-rack ruby-rack-protection ruby-sinatra
> ```
> 
> As per: [#1628 (comment)](https://github.com/beefproject/beef/issues/1628#issuecomment-464289170)

3q
